airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py

```diff
@@ -2,25 +2,202 @@
 # Copyright (c) 2021 Airbyte, Inc., all rights reserved.
 #
 
+import copy
 import logging
-from typing import Any, MutableMapping
+from typing import Any, Iterator, List, Mapping, MutableMapping, Optional, Tuple
 
-from airbyte_cdk.sources.deprecated.base_source import BaseClient, BaseSource, ConfiguredAirbyteStream
+from airbyte_cdk.models import AirbyteMessage, ConfiguredAirbyteCatalog
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.deprecated.base_source import ConfiguredAirbyteStream
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.utils.schema_helpers import InternalConfig, split_config
+from airbyte_cdk.utils.event_timing import create_timer
+from requests import HTTPError
+from source_hubspot.api import (
+    API,
+    Campaigns,
+    ContactLists,
+    ContactsListMemberships,
+    CRMObjectIncrementalStream,
+    CRMSearchStream,
+    DealPipelines,
+    Deals,
+    EmailEvents,
+    Engagements,
+    Forms,
+    FormSubmissions,
+    MarketingEmails,
+    Owners,
+    PropertyHistory,
+    SubscriptionChanges,
+    TicketPipelines,
+    Workflows,
+)
 
-from .client import Client
 
+class SourceHubspot(AbstractSource):
+    def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
+        """Check connection"""
+        alive = True
+        error_msg = None
+        common_params = self.get_common_params(config=config)
+        try:
+            contacts = CRMSearchStream(
+                entity="contact", last_modified_field="lastmodifieddate", associations=["contacts"], name="contacts", **common_params
+            )
+            _ = contacts.properties
+        except HTTPError as error:
+            alive = False
+            error_msg = repr(error)
 
-class SourceHubspot(BaseSource):
-    client_class = Client
+        return alive, error_msg
 
-    def _read_stream(
-        self, logger: logging.Logger, client: BaseClient, configured_stream: ConfiguredAirbyteStream, state: MutableMapping[str, Any]
-    ):
+    @staticmethod
+    def get_api(config: Mapping[str, Any]) -> API:
+        credentials = config.get("credentials", {})
+        return API(credentials=credentials)
+
+    def get_common_params(self, config) -> Mapping[str, Any]:
+        start_date = config.get("start_date")
+        api = self.get_api(config=config)
+        common_params = dict(api=api, start_date=start_date)
+        return common_params
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        credentials = config.get("credentials", {})
+        common_params = self.get_common_params(config=config)
+        streams = [
+            Campaigns(**common_params),
+            CRMSearchStream(
```

Comment:
@augan-rymkhan I propose to follow here the practice we have in many other connectors:
make these attribute of class and create explicit classes for each stream, i.e.
```python
class Contacts(HubspotCRMSearchStream):
   entity = "contact"
   cursor_field = "lastmodifieddate"
   associations=["contacts"]
```